src/OrleansProviders/Streams/Common/Monitors/ICacheMonitor.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Responsible for monitoring cache related metrics
+    /// </summary>
+    public interface ICacheMonitor
+    {
+        #region Event driven metrics
+
+        /// <summary>
+        /// Track cache pressure metrics when cache pressure monitor encounter a status change
+        /// </summary>
+        void TrackCachePressureMonitorStatusChange(string pressureMonitorType, bool underPressure, double? cachePressureContributionCount, double? currentPressure);
+
+        /// <summary>
+        /// Track message added to the cache, and newest and oldest messsges among them 
+        /// </summary>
+        /// <param name="messageAdded"></param>
+        void TrackMessageAdded(long messageAdded);
+
+        /// <summary>
+        /// Track message purged from the cache, and the newest and oldest messages among them
+        /// </summary>
+        /// <param name="messagePurged"></param>
+        void TrackMessagePurged(long messagePurged);
+
+        /// <summary>
+        /// Track new memory allocated by the cache
+        /// </summary>
+        /// <param name="memoryInByte"></param>
+        void TrackMemoryAllocated(int memoryInByte);
```

Comment:
> why the metric can be easily misunderstood.

This is the memory allocated to the cache, not necessarily the memory use, nor does it necessarily accurately account for the size of the messages used by the cache.

For instance if the cache reports having 1k messages and using 1mb of memory, a maintainer may reseasonlby conclude that the average message size is 1k, but the 1mb includes the message payload, and other evenhub data we track plus space lost by fragmentation, so the actual message size may be half that.  This is not a big deal really, I was just pointing out that we've a metric with limited utility (IMO) that is also somewhat ambiguous.  I'd not include it for these reasons, but I won't protest if you feel differently.

>TrackMemoryNewlyAllocated

I think current name is fine.